### PR TITLE
Added Rspec for enumerable methods

### PR DESCRIPTION
**Pull request by bettercallazamat**
We added Rspec for enumerable methods following the instructions required by [Microverse Curriculum ](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/).

Any new changes will be documented and pushed on this pull request.

### CODE REVIEW

File: spec/main_spec.rb [hunk 1]
```diff
@@ -0,0 +1,184 @@
+# spec :main_spec.rb
+require './main.rb'
+
+RSpec.describe Enumerable do
+  let(:array) { [11, 2, 3, 56] }
+
```
Comment (majovanilla) [inline spec/main_spec.rb]: You should test other types of enumerables: ranges, array of strings, a block of code.

File: spec/main_spec.rb [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  end
+
+  describe '.my_each_with_index' do
+    context 'when receive a block' do
+      it 'push each value to the result array with index' do
+        result = []
+        array.my_each_with_index { |x, y| result.push([x, y]) }
+        expect(result[0][1]).to(eq(0))
+      end
+    end
```
Comment (majovanilla) [inline spec/main_spec.rb]: Include a test to check if the index it's working correctly.

File: spec/main_spec.rb [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  end
+
+  describe '.my_select' do
+    context 'when receive a block' do
+      it 'returns selected values' do
+        result = array.my_select { |x| x > 5 }
+        expect(result).to eq([11, 56])
+      end
+    end
```
Comment (majovanilla) [inline spec/main_spec.rb]: Try this with other types of enumerable as suggested.

File: spec/main_spec.rb [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    context 'when no block is given' do
+      it 'returns true if one or more values are truthy' do
+        expect([nil, true, 99].my_any?).to eq true
+      end
+
+      it 'returns false if one or more values are falsy' do
+        expect([].my_any?).to eq false
+      end
+    end
+  end
```
Comment (majovanilla) [inline spec/main_spec.rb:133]: Good job considering negative cases

File: spec/main_spec.rb [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+        expect(array.my_count(&:even?)).to eq 2
+      end
+    end
+
+    context 'when no argument is given' do
+      it 'returns number of values' do
+        expect(array.my_count).to eq 4
+      end
+    end
+  end
```
Comment (majovanilla) [inline spec/main_spec.rb]: Include a test to check the negative case.

File: spec/main_spec.rb [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+        expect(array.my_inject { |sum, n| sum + n }).to eq 72
+      end
+    end
+
+    context 'when receive a symbol' do
+      it 'combines all elements of enum by applying a binary operation, specified by a symbol' do
+        expect(array.my_inject(1, :*)).to eq 3_696
+      end
+    end
+  end
```
Comment (majovanilla) [inline spec/main_spec.rb]: Chech this method with other types of enumerable.

**Review by majovanilla — CHANGES_REQUESTED** [review]
Hey team, 👋

## You did an *amazing*🌟 ✨ job on this project! 👊🏻  💯
There are still some changes 👇 you need to implement to approve this request 🔍

### Required Changes: ✍️

- Include the instructions to test your files on your ReadMe
- Check the comments above.
## Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack (@majo)
I would be happy to help! 👩‍💻 


Happy coding 🤓

--Maria Reyes

File: spec/main_spec.rb [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+  let(:array_of_strings) { %w[a b c d] }
+  let(:range) { 5..8 }
+
+  describe '.my_each' do
+    context 'when receive a block' do
+      it 'push each value to the result array' do
+        result = []
+        array.my_each { |x| result.push(x) }
+        expect(result).to(eq(array))
+      end
```
Comment (shshamim63) [inline spec/main_spec.rb:15]: [Optional]
We want our my_each to work exactly to each method right? So why not check it with it? We can transform the test case something like this->
```ruby
it 'push each value to the result array' do
    expected = []
    array.my_each { |x| result.push(x) }
    current_result = expected.dup
    array.each { |x| result.push(x) }
    expect(current_result).to eq(expected)
end
```

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## Status :1234:  Changes Required :1234: 
___
Hi @bettercallazamat and @ghamtre :wave: ,
Awesome job :clap: You have covered all the changes asked by the previous review except one major one. Which is currently asking blocking me to approve this project now. Check the comment below.
___
### :spades: Mandatory Changes :spades: 
___
1. As the previous reviewer mentioned, still I cant not see any instruction about how to execute the test cases. Now the question is why I should add it? The first thing a user looks at a repository is the readme.  So if it does not contain all the core information related to the project, the user will lose interest. So the core features of this project are enumerable methods and RSpec. So it will be great if you add it to the readme.
___
### :clubs: Optional Changes :clubs: 
___
1. You can check your methods with the actual methods. I have added an example. Obviously there are thousands of ways to cover it. But comparing with the actual seems more valid to me.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Review by stratospherique — APPROVED** [review]
### Great Job implementing the suggested changes :clap: :clap: Congrats, your project is approved :heavy_check_mark: You may merge :+1: :+1: 

Keep up the good work :muscle: 
**_Ahmed Mahfoudh_**
